registry/registry.go

```diff
@@ -49,7 +49,7 @@ func (r *registrar) NamePrefix() string {
 // Find looks at its internal index to find a registration that matches the
 // entity instance provided. Return an error when not found.
 func (r *registrar) Find(entity dosa.DomainObject) (*dosa.RegisteredEntity, error) {
-	name := reflect.TypeOf(entity).Name()
+	name := reflect.TypeOf(entity).Elem().Name()
```

Comment:
This needs a test. It could never have worked. Additionally, test the case where they have the same entity name in two different packages, which I think will fail because you are stripping the package part.

This is unlikely, but possible, since they can rename the underlying table using the `name=` attribute.